### PR TITLE
Predict dataset

### PR DESCRIPTION
**Pull request by AndyMcAliley**
This PR adds Snakemake targets for unnormalizing training, validation, and test sets and making predictions. Unnormalized data and predictions are computed and saved to npz files.

# How to run the code
The rule `unnormalized_dataset` computes and saves unnormalized dynamic features, static features, and observations for one dataset: training, validation, or testing. 

The rule `predict_dataset` computes and saves predictions for the training, validation, or testing set. 
> NOTE: It requires a compiled dataset and a trained model, each of which takes several hours to obtain. If you want, copy the following files from Tallgrass, preserving their directory structure relative to the repo root directory:
> - 2_process/out/model_prep/valid.npz
> - 3_train/out/model_prep/initial/local_a_metadata.npz
> - 3_train/out/model_prep/initial/local_a_weights.npz

To run this part of the pipeline:
```
snakemake -c1 -s 4_evaluate.smk 4_evaluate/out/model_prep/initial/local_a/unnormalized_valid.npz
```
That should save unnormalized validation data to `4_evaluate/out/model_prep/initial/local_a/unnormalized_valid.npz`.
```
snakemake -c1 -s 4_evaluate.smk 4_evaluate/out/model_prep/initial/local_a/preds_valid.npz
```
That should save predictions for the validation set to `4_evaluate/out/model_prep/initial/local_a/preds_valid.npz`.

# How to review this PR

## Level of review requested

The main things I'd like reviewed are:

1. Do the pipeline additions and modifications make sense?
2. Have any errors been introduced into the pipeline?

## Where in the code to focus

The three new files that have been created: `4_evaluate.smk`, `unnormalize_dataset.py`, and `predict_dataset.py`.

## Issues that will be addressed in upcoming PRs
- Compute metrics: Additional functions will compute RMSE and bias.
- Make plots: plotting functions will use the saved predictions so that they do not need to recompute them

### CODE REVIEW

**Comment by AndyMcAliley** [comment]
Thanks! I realized that `unnormalized_dataset()` doesn't need
```
    # Get number of inputs/outputs
    n_depths = len(metadata['depths_use'])
    n_dynamic = len(metadata['dynamic_features_use'])
    n_static = len(metadata['static_features_use'])
```
so I removed that code.
